src/core/Akka.Remote/Transport/DotNetty/AkkaLoggingHandler.cs

```diff
@@ -8,14 +8,93 @@
 using System;
 using System.Net;
 using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 using Akka.Util;
 using DotNetty.Buffers;
+using DotNetty.Common.Concurrency;
 using DotNetty.Transport.Channels;
 using ILoggingAdapter = Akka.Event.ILoggingAdapter;
 
 namespace Akka.Remote.Transport.DotNetty
 {
+    internal class BatchWriter : ChannelHandlerAdapter
+    {
+        private readonly int _maxPendingWrites;
+        private readonly long _maxPendingMillis;
+
+        public BatchWriter(int maxPendingWrites = 30, long maxPendingMillis = 40l)
+        {
+            _maxPendingWrites = maxPendingWrites;
+            _maxPendingMillis = maxPendingMillis;
+        }
+
+        private int _currentPendingWrites = 0;
+
+        public bool HasPendingWrites => _currentPendingWrites > 0;
+
+        public override void HandlerAdded(IChannelHandlerContext context)
+        {
+            ScheduleFlush(context);
+            base.HandlerAdded(context);
+        }
+
+        public override Task WriteAsync(IChannelHandlerContext context, object message)
+        {
+            var write = base.WriteAsync(context, message);
+            if (++_currentPendingWrites == _maxPendingWrites)
+            {
+                context.Flush();
+                Reset();
+            }
+
+            return write;
+        }
+
+        void ScheduleFlush(IChannelHandlerContext context)
+        {
+            // Schedule a recurring flush - only fires when there's writable data
+            var time = TimeSpan.FromMilliseconds(_maxPendingMillis);
```

Comment:
By default, we check to see if messages need to be flushed every 40ms - this is designed for very low-volume systems where they'll probably never meet the msgCount / max bytes threshold I set earlier.